### PR TITLE
Add GitHub Issue Templates

### PR DESCRIPTION
**Pull request by CloudNiner**
For:
- Bug Reports
- Feature Requests
- New Releases

Mostly this was a way to document our release process and capture
it in issues, similar to other Azavea OSS projects, until we are able to
address https://github.com/locationtech/geotrellis/issues/3140

Bonus that we can begin iterating on a few other common issue
template types.

@pomadchin pay particular attention to the steps in the new release.md issue template. Please double-check that I didn't miss any of the key release steps.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+A clear and concise description of what the bug is.
+
+## To Reproduce
+
+Steps to reproduce the behavior:
+
+1. Go to '...'
+2. Click on '....'
+3. Scroll down to '....'
+4. See error
```
Comment (pomadchin) [inline .github/ISSUE_TEMPLATE/bug_report.md:21]: I would say that we're mostly interested in code examples, inputs for the code and produced outputs.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+---
+
+## Is your feature request related to a problem? Please describe.
+
+A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
+
+## Describe the solution you'd like
+
+A clear and concise description of what you want to happen.
```
Comment (pomadchin) [inline .github/ISSUE_TEMPLATE/feature_request.md:16]: Some pseudocode with expected inputs and outputs can be helpful as well, to undestand the context better.

File: .github/ISSUE_TEMPLATE/release.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+[ ] Create a new tag at the previous commit with `git tag -a vX.Y.Z -m "vX.Y.Z"`
+
+[ ] Push the tag to `locationtech/geotrellis` with `git push --tags`
+
+[ ] Follow the instructions in the setup section of [publish/README.md](https://github.com/locationtech/geotrellis/blob/master/publish/README.md#setup) to ensure that your local environment is configured for publishing the release
+
+[ ] Set the `RELEASE_TAG` variable in `./publish/Makefile` to `vX.Y.Z`, substituting x/y/z for the current release version
+
+[ ] **Before Continuing:** Ensure you have pushed the tag you created in step 3 before continuing. Otherwise the next step will fail.
```
Comment (pomadchin) [inline .github/ISSUE_TEMPLATE/release.md:22]: I guess steps should be numbered?

Comment (CloudNiner) [reply .github/ISSUE_TEMPLATE/release.md:22]: Looks like we haven't done this for other release templates and the markdown is readable without. I'd prefer to remain consistent with other Azavea projects unless there's a clear win we get for adding them.

**Comment by pomadchin** [comment]
@CloudNiner oh, I don't know how would it looks like; if it is not convenient, why not?

**Comment by CloudNiner** [comment]
Created config.yml with Gitter link instead of adding another issue template. Kept the ability for users to open blank issues because there's definitely issue categories we aren't accounting for.

**Pull request merged**